src/webnn_native/ops/Binary.cpp

```diff
@@ -19,6 +19,76 @@
 
 namespace webnn_native { namespace op {
 
+    MaybeError Binary::CalculateShape() {
+        auto inputShape1 =
+            mInputs[0]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[0]->Shape();
+        auto inputShape2 =
+            mInputs[1]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[1]->Shape();
+
+        auto l1 = inputShape1.size(), l2 = inputShape2.size();
+        bool shape1IsBigger = l1 >= l2;
+        auto maxShape = shape1IsBigger ? inputShape1 : inputShape2;
+        auto minShape = shape1IsBigger ? inputShape2 : inputShape1;
+        std::vector<int32_t> outputShape;
+        if (mOpType == kMatMul) {
```

Comment:
Given there is no much code sharing between MatMul and element-wise binary ops, I would suggest to split them into two functions.

Reply:
Yes, thanks, but the matmul has been implemented as a case of Binary, I am not clear how to split into two functions, could you give more details?

Reply:
Probably `CaculateMatMulShape` and `CaculateElementWiseBinaryShape`.